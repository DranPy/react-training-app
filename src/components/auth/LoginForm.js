import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';

import InputWithLabel from 'components/forms/fields/InputWithLabel';

class LoginForm extends Component {
  static propTypes = {
    handleSubmit: PropTypes.func,
    valid: PropTypes.bool,
    submitting: PropTypes.bool,
    errorMessage: PropTypes.string,
    isLoading: PropTypes.bool,
    isAuthorized: PropTypes.bool,
  };

  errorMessage = () => {
    const { errorMessage } = this.props;
    if (errorMessage) {
      return <div className="alert alert-warning">{errorMessage}</div>;
    }
  };

  render() {
    const { submitting, handleSubmit, valid } = this.props;

    return (
      <div className="container" style={{ width: 400 }}>
        <form onSubmit={handleSubmit}>
          <Field
            label="Email"
            className="form-control"
            name="email"
            component={InputWithLabel}
            type="email"
          />
          <Field
            label="Password"
            className="form-control"
            name="password"
            component={InputWithLabel}
            type="password"
          />
          <div>
            <input
              type="submit"
              className="btn btn-block btn-primary"
              disabled={submitting || !valid}
              value="Login"
            />
          </div>
          {this.errorMessage()}
        </form>
      </div>
    );
  }
}

export default LoginForm;
