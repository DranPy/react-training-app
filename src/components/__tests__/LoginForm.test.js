import { shallow } from 'enzyme';
import LoginForm from '../LoginForm';
import React from 'react';

describe('<LoginForm />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<LoginForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should submit form', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<LoginForm handleSubmit={handleSubmit} />);
    wrapper.find('form').simulate('submit');
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });
});
