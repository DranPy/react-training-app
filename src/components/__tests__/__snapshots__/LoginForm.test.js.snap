// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should match snapshot 1`] = `
<div
  className="container"
>
  <form>
    <div
      className="form-group row"
    >
      <div
        className="col-4 col-form-label"
      >
        <label
          className="form-control-plaintext"
        >
          Email
        </label>
      </div>
      <div
        className="col-8"
      >
        <Field
          className="form-control"
          component="input"
          name="email"
          type="email"
        />
      </div>
      <div
        className="col-4 col-form-label"
      >
        <label
          className="form-control-plaintext"
        >
          Password
        </label>
      </div>
      <div
        className="col-8"
      >
        <Field
          className="form-control"
          component="input"
          name="password"
          type="password"
        />
      </div>
    </div>
    <div>
      <input
        className="btn btn-sm"
        type="submit"
        value="Login"
      />
    </div>
  </form>
</div>
`;
